specifications/config.json

```diff
@@ -0,0 +1,127 @@
+{
+  "client": {
+    "apiKey":{
+      "file":"",
+      "id":"",
+      "secret":""
+    },
+    "cacheManager":{
+      "defaultTtl": 300,
+      "defaultTti": 300,
+      "caches": {
+        "account": {
+          "ttl": 300,
+          "tti": 300
+        }
+      }
+    },
+    "baseUrl": "https://api.stormpath.com/v1",
+    "connectionTimeout": 30,
+    "authenticationScheme": "SAUTHC1", // or "BASIC"
+    "proxy":{
+      "port": "",
+      "host": "",
+      "username": "",
+      "password": ""
+    }
+  },
+  "application": {
+    "name": "test",
+    "href": "https://..."
+  },
+  "web": {
+    "oauth2":{
+      "enabled": false,
+      "uri": "/oauth/token",
+      "password":{
+        "enabled": true,
+        "accessToken":{
+          "ttl": 3600
+        }
+      },
+      "client_credentials":{
+        "enabled": true,
+        "accessToken":{
+          "ttl": 3600
+        }
+      }
+    },
+    "accessTokenCookie":{
+      "name": "access_token",
+      "ttl": 3600,
+      "tti": 900,
+      "httpOnly": true,
+      "secure": null,
+      "path": "/",
+      "domain": null
+    },
+    "register": {
+      "enabled": false,
+      "uri": "/register",
+      "nextUri": "/",
+      "autoAuthorize": false,
+      "fields": [
```

Comment:
Thought:

If we go with a map of fieldName-to-fieldConfig pairs, I think ordering might be better represented as an array of names, rather than an index field.  For example:

```json
{
  "fields": {
    "givenName": { ... etc ... },
   ... etc ...
  },
  "displayOrder": ["givenName", "surname", "email", "password", "passwordConfirm"]
}
```

This way you have one place to do the updating rather than going back and forth and changing multiple definitions and ensuring the numbers are correct.  The Java SDK works like this and it's really nice from a usability perspective.

Thoughts?